beetsplug/web/__init__.py

```diff
@@ -225,10 +226,22 @@ def item_file(item_id):
     else:
         item_path = util.py3_path(item.path)
 
+    try:
+        unicode_item_path = util.text_string(item.path)
+    except (UnicodeDecodeError, UnicodeEncodeError):
+        unicode_item_path = u"fallback" + os.path.splitext(item_path)[1]
+
+    try:
+        # Imitate http.server behaviour
+        os.path.basename(unicode_item_path).encode("latin-1", "strict")
+        safe_filename = os.path.basename(unicode_item_path)
```

Comment:
This line should probably go in an `else` case for the try/catch block, which runs when an exception does not occur.